File: ios-app/Packages/BayitCore/Sources/BayitCore/Discover/DiscoverFeature.swift
import Foundation

public enum Platform: String, Sendable {
    case iOS
    case tvOS
}

public struct DiscoverFeature: Identifiable, Sendable {
    public let id: String
    public let category: DiscoverCategory
    public let nameKey: String
    public let taglineKey: String
    public let descriptionKey: String
    public let iconName: String
    public let platforms: Set<Platform>
    public let prerequisites: [FeaturePrerequisite]
    public let walkthroughSteps: [WalkthroughStep]
    public let deepLinkRoute: String?

    public init(
        id: String,
        category: DiscoverCategory,
        nameKey: String,
        taglineKey: String,
        descriptionKey: String,
        iconName: String,
        platforms: Set<Platform>,
        prerequisites: [FeaturePrerequisite],
        walkthroughSteps: [WalkthroughStep],
        deepLinkRoute: String? = nil
    ) {
        self.id = id
        self.category = category
        self.nameKey = nameKey
        self.taglineKey = taglineKey
        self.descriptionKey = descriptionKey
        self.iconName = iconName
        self.platforms = platforms
        self.prerequisites = prerequisites
        self.walkthroughSteps = walkthroughSteps
        self.deepLinkRoute = deepLinkRoute
    }
}

File: android-app/feature/feature-discover/src/main/java/tv/bayit/plus/feature/discover/model/DiscoverFeature.kt
package tv.bayit.plus.feature.discover.model

data class DiscoverFeature(
    val id: String,
    val category: DiscoverCategory,
    val nameKey: String,
    val taglineKey: String,
    val descriptionKey: String,
    val iconName: String,
    val platforms: Set<DiscoverPlatform>,
    val prerequisites: List<FeaturePrerequisite>,
    val walkthroughSteps: List<WalkthroughStep>,
    val deepLinkRoute: String? = null,
)

enum class DiscoverPlatform {
    IOS,
    TVOS,
    ANDROID,
}

data class FeaturePrerequisite(
    val id: String,
    val type: PrerequisiteType,
    val labelKey: String,
    val fixRoute: String? = null,
)

enum class PrerequisiteType {
    AVATAR,
    SUBSCRIPTION,
    MICROPHONE,
    CONTENT_TYPE,
    PREFERENCE,
    VOICE_CLONE,
}

data class WalkthroughStep(
    val id: String,
    val instructionKey: String,
    val targetAccessibilityId: String,
    val expectedAction: WalkthroughAction,
    val order: Int,
)

enum class WalkthroughAction {
    TAP,
    NAVIGATE,
    PAUSE,
    SELECT,
    TYPE,
    OBSERVE,
    CREATE_AVATAR,
}
